lib/Onyx.js

```diff
@@ -1053,18 +1054,18 @@ function hasPendingMergeForKey(key) {
  * Otherwise removes all nested null values in objects and returns the object
  * @param {String} key
  * @param {Mixed} value
- * @returns {Mixed} `null` if the key got removed completely, otherwise the value without null values
+ * @returns {Mixed} The value without null values and a boolean "wasRemoved", which indicates if the key got removed completely
  */
 function removeNullValues(key, value) {
     if (_.isNull(value)) {
         remove(key);
-        return null;
+        return {value, wasRemoved: true};
```

Comment:
NAB: It seems pointless to have this `wasRemoved` flag when the caller could just check if the value is null to determine if the key was removed.

Reply:
The reason i introduced this extra flag was to have a consistent variable/condition of whether the key got removed as part of an operation. We check for this in multiple spots (both of your other comments), and i felt that having a dedicated `wasRemoved` flag was better than always checking for `if (valueAfterRemoving === null)`.

Reply:
I generally prefer limiting the number of parameters. Also, wasRemoved is always simply `_.isNull(value)`, except in one case where we pass the batched changes. In fact you have a comment there explaining that we're checking if the batched changes are null. That's kind of silly.

Reply:
I'll leave it up to you if you want to change this is a follow up PR.